otx/algorithms/common/adapters/mmcv/models/backbones/torchvision_backbones.py

```diff
@@ -0,0 +1,241 @@
+        if name == "bn":
+            model._modules[name] = build_norm_layer(cfg, num_features=module.num_features)[1]
+    return model
+
+
+def resnet_forward(self, x):
+    """Resnet forward function for wrapping model (refer to torchvision)."""
+    outputs = []
+    y = x
+    stages = [self.layer1, self.layer2, self.layer3, self.layer4]
+    last_stage = max(self.out_indices)
+    y = self.conv1(y)
+    y = self.bn1(y)
+    y = self.relu(y)
+    y = self.maxpool(y)
+    for i, stage in enumerate(stages):
+        y = stage(y)
+        if i in self.out_indices:
+            outputs.append(y)
+        if i == last_stage:
+            break
+    return outputs
+
+
+def shufflenet_forward(self, x):
+    """Shufflenet forward function for wrapping model (refer to torchvision)."""
+    outputs = []
+    y = x
+    y = self.conv1(y)
+    y = self.maxpool(y)
+    stages = [self.stage2, self.stage3, self.stage4, self.conv5]
+    last_stage = max(self.out_indices)
+    for i, stage in enumerate(stages):
+        y = stage(y)
+        if i in self.out_indices:
+            outputs.append(y)
+        if i == last_stage:
+            break
+    return outputs
+
+
+def multioutput_forward(self, x):
+    """Multioutput forward function for new model (copy from mmdet)."""
+    outputs = []
+    y = x
+
+    last_stage = max(self.out_indices)
+    if hasattr(self, "features"):
+        stages = self.features
+    elif hasattr(self, "layers"):
+        stages = self.layers
+    else:
+        raise ValueError(f"Not supported multioutput forward: {self}")
+
+    for i, stage in enumerate(stages):
+        y = stage(y)
+        temp_s = str(i) + " " + str(y.shape)
+        if i in self.out_indices:
+            outputs.append(y)
+            temp_s += "*"
+        if self.verbose:
+            print(temp_s)
+        if i == last_stage:
+            break
+    return outputs
+
+
+def train(self, mode=True):
+    """Train forward function for new model (copy from mmdet)."""
+    super(self.__class__, self).train(mode)
+
+    if hasattr(self, "features"):
+        stages = self.features
+    elif hasattr(self, "layers"):
+        stages = self.layers
+    else:
+        raise ValueError(f"Not supported multioutput forward: {self}")
+
+    for i in range(self.frozen_stages + 1):
+        temp_m = stages[i]
+        temp_m.eval()
+        for param in temp_m.parameters():
+            param.requires_grad = False
+
+    if mode and self.norm_eval:
+        for mmodule in self.modules():
+            # trick: eval have effect on BatchNorm only
+            if isinstance(mmodule, _BatchNorm):
+                mmodule.eval()
+
+
+def init_weights(self, pretrained=True):
+    """Init weights function for new model (copy from mmdet)."""
+    if pretrained and self.model_urls:
+        state_dict = load_state_dict_from_url(self.model_urls)
+        self.load_state_dict(state_dict)
+
+
+def generate_torchvision_backbones():
```

Comment:
it's just a question. why don't you just use inheritance? then, you just need to override some of attributes.

Reply:
This is the code copied from `mmdetection` to register the `pytorchcv` backbone in the registry. (I modified the existing code for torchvision.) I tried to clean it up as much as possible, but the inheritance still remains. 
If mmdetection changes to public later and pytorchcv comes to the otx side, I think it's good to fix it together. What do you think?

Reply:
I understand. let's refactor together later then.